model/post.go

```diff
@@ -246,3 +247,17 @@ func PostPatchFromJson(data io.Reader) *PostPatch {
 
 	return &post
 }
+
+var channelMentionRegexp = regexp.MustCompile(`(^|\s)(~([a-zA-Z0-9.\-_]*))`)
+
+func (o *Post) ChannelMentions() (names []string) {
```

Comment:
We can save some time here by just checking `strings.Contains(o.Message, "~")` before using the regex. That helped a fair bit when doing some similar preprocessing to stop notifying people mentioned in code blocks